custodia/httpd/server.py

```diff
@@ -85,6 +86,42 @@ def server_bind(self):
         self.socket_file = self.socket.getsockname()
 
 
+class ForkingTLSServer(ForkingHTTPServer):
+    def __init__(self, server_address, handler_class, config, sslcontext=None):
+        ForkingHTTPServer.__init__(self, server_address, handler_class, config)
+        if sslcontext is not None:
+            self.sslcontext = sslcontext
+        else:
+            self.sslcontext = self.mksslcontext()
+
+    def mksslcontext(self):
+        certfile = self.config.get('ssl_certfile')
+        keyfile = self.config.get('ssl_keyfile')
+        cafile = self.config.get('ssl_cafile')
+        capath = self.config.get('ssl_capath')
+        verifymode = self.config.get('ssl_verifymode', 'CERT_NONE')
```

Comment:
That is the server context. CERT_NONE disables client certs. With CERT_REQUIRED the client has to provide a valid client certificate that is signed by a trusted CA.

The client side always requires the server to have valid cert.